homeassistant/components/proximity/config_flow.py

```diff
@@ -89,11 +89,19 @@ async def async_step_user(
         if user_input is not None:
             self._async_abort_entries_match(user_input)
 
-            zone = self.hass.states.get(user_input[CONF_ZONE])
-
-            return self.async_create_entry(
-                title=cast(State, zone).name, data=user_input
-            )
+            title = cast(State, self.hass.states.get(user_input[CONF_ZONE])).name
+
+            existing_entry_titles = [
+                e.title.lower() for e in self._async_current_entries()
+            ]
+            if any(t for t in existing_entry_titles if t == title.lower()):
+                for possible_title in [f"{title}{i}" for i in range(2, 11)]:
```

Comment:
Why don't we use similar code as when we generate an entity id?

https://github.com/home-assistant/core/blob/630c8b28ca4d4343a040aa72bd9b88f630467187/homeassistant/helpers/entity_registry.py#L605

Reply:
because... I wasn't aware of this 🙈

Reply:
unfortunately this function does not suits here, since it searches for existing entity_ids in the entity registry, but we need to find a "free" config entry title

Reply:
I didn't mean that we should use that function here. I meant that we should use the same logic. We should eg not limit ourselves to 10 attempts.

Reply:
ahhh ... makes sense 😁